geode-core/src/main/java/org/apache/geode/management/internal/cli/util/ClasspathScanLoadHelper.java

```diff
@@ -19,26 +19,47 @@
 import java.util.Arrays;
 import java.util.Collection;
 import java.util.Set;
-import java.util.regex.Pattern;
 
 import io.github.classgraph.ClassGraph;
 import io.github.classgraph.ClassInfoList;
 import io.github.classgraph.ScanResult;
-import io.github.classgraph.utils.WhiteBlackList;
 
 /**
  * Utility class to scan class-path & load classes.
  *
  * @since GemFire 7.0
  */
-public class ClasspathScanLoadHelper {
+public class ClasspathScanLoadHelper implements AutoCloseable {
 
   private final ScanResult scanResult;
 
   public ClasspathScanLoadHelper(Collection<String> packagesToScan) {
     scanResult = new ClassGraph().whitelistPackages(packagesToScan.toArray(new String[] {}))
         .enableClassInfo()
-        .enableAnnotationInfo().scan();
+        .enableAnnotationInfo().scan(optimalNumScanThreads());
+  }
+
+  /**
+   * @return a safe number of classgraph scan threads
+   */
+  private static int optimalNumScanThreads() {
```

Comment:
I really don't like 'magic' code like this. In general, heuristics like this take some testing to get right and typically we'll never cover every eventuality. That means they need to allow overriding by setting some kind of system property which is also not ideal. In this particular case (and from the testing I've done with Classgraph) the number of threads has no correlation either with memory consumption or open file descriptors. Other than this bit, the rest LGTM.

Reply:
Would setting it to "2" simplify the logic and runtime expectations?

Reply:
fair enough @jdeppe-pivotal, it's "magic code like this" is classgraph that prompted this change in the first place.  I would even go a step beyond @metatype's suggestion and just lock it to "1"

Reply:
changed to always use 1 thread